Assets/_Core/Scripts/Internal/EntitySystem/FilterRules.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+
+public enum TagFilterType
+{
+	None,
+	HasAnyTag,
+	HasAllTags,
+}
+
+public class FilterRules
+{
+	public TagFilterType FilterType
+	{
+		get; private set;
+	}
+
+	public string[] FilterTags
+	{
+		get
+		{
+			return _filterTags.ToArray();
+		}
+	}
+
+	public Type[] FilterComponents
+	{
+		get
+		{
+			return _componentsToFilterOn.ToArray();
+		}
+	}
+
+	private List<string> _filterTags = new List<string>();
+	private List<Type> _componentsToFilterOn = new List<Type>();
+
+	private static bool _filterOpened = false;
+	private static FilterRules _constructingFiltersParameters;
+
+	public static void OpenFilterCreation()
```

Comment:
I recommend adding documentation to this functionality.

Reply:
Fixed in https://github.com/Darkfafi/TextShooter/pull/13/commits/f83d457fe0bc64deaf4e30eab8c10ac2b0076996